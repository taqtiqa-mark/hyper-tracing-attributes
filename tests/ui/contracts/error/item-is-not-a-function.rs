use http_tracing_attributes::contracts;

#[contracts]
struct S;

fn main() {}
