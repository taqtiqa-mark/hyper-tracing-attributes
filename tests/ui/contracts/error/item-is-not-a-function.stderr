warning: unused variable: `meta_args`
  --> src/lib.rs
   |
   |     let mut meta_args = syn::parse_macro_input!(meta as crate::server_send::parse::Args);
   |             ^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_meta_args`
   |
   = note: `#[warn(unused_variables)]` on by default

warning: variable does not need to be mutable
  --> src/lib.rs
   |
   |     let mut meta_args = syn::parse_macro_input!(meta as crate::server_send::parse::Args);
   |         ----^^^^^^^^^
   |         |
   |         help: remove this `mut`
   |
   = note: `#[warn(unused_mut)]` on by default

error: item is not a function

         = help: `#[contracts]` can only be used on functions

 --> tests/ui/contracts/error/item-is-not-a-function.rs:4:1
  |
4 | struct S;
  | ^^^^^^^^^
